src/engine/Service.cpp

```diff
@@ -323,19 +323,18 @@ std::optional<std::string> Service::getSiblingValuesClause() const {
   checkCancellation();
 
   ad_utility::HashSet<std::string> rowSet;
-
   std::string values = " { ";
   for (size_t rowIndex = 0; rowIndex < siblingResult->idTable().size();
        ++rowIndex) {
     std::string row = "(";
     for (const auto& columnIdx : commonColumnIndices) {
-      const auto& optionalString = ExportQueryExecutionTrees::idToStringAndType(
+      const auto& optionalString = idToValueForValuesClause(
           siblingTree_->getRootOperation()->getIndex(),
           siblingResult->idTable()(rowIndex, columnIdx),
           siblingResult->localVocab());
 
       if (optionalString.has_value()) {
-        absl::StrAppend(&row, optionalString.value().first, " ");
+        absl::StrAppend(&row, optionalString.value(), " ");
```

Comment:
I think this is not quite correct:
You can only get `nullopt` if there are blank nodes in the input. And in that case, you should drop the complete row, as it is not needed, because it will never lead to any bindings. In particular the replacing by `""` does not work, as you will end up with a VALUES clause that will be invalid and rejected by the remote parser, because there are values missing in some of the rows.